dotCMS/src/integration-test/java/com/dotmarketing/portlets/variant/business/VariantFactoryTest.java

```diff
@@ -0,0 +1,411 @@
+                DigestUtils.sha256Hex(variant.getName()), variantSaved.getIdentifier());
+    }
+
+    /**
+     * Method to test: {@link VariantFactory#save(Variant)}
+     * When: Try to save a {@link Variant} object without name
+     * Should: throw {@link NullPointerException}
+     *
+     * @throws DotDataException
+     */
+    @Test(expected = IllegalArgumentException.class)
+    public void saveWithoutName() throws DotDataException {
+        final Variant variant = new Variant("1", null, false);
+        FactoryLocator.getVariantFactory().save(variant);
+    }
+
+    /**
+     * Method to test: {@link VariantFactory#update(Variant)}
+     * When: Try to update a {@link Variant} object
+     * Should: Update it in Data base.
+     *
+     * @throws DotDataException
+     */
+    @Test
+    public void update() throws DotDataException {
+        final Variant variant = new VariantDataGen().next();
+
+        final Variant variantSaved = FactoryLocator.getVariantFactory().save(variant);
+
+        assertNotNull(variantSaved);
+        assertNotNull(variantSaved.getIdentifier());
+
+        final Variant variantUpdated = new Variant(variantSaved.getIdentifier(),
+                variantSaved.getName() + "_updated", false);
+
+        FactoryLocator.getVariantFactory().update(variantUpdated);
+
+        final Variant variantFromDataBase = getVariantFromDataBase(variantSaved);
+
+        assertEquals(variantUpdated.getName(), variantFromDataBase.getName());
+        assertEquals(variantUpdated.getIdentifier(), variantFromDataBase.getIdentifier());
+        assertFalse(variantFromDataBase.isArchived());
+    }
+
+    /**
+     * Method to test: {@link VariantFactory#update(Variant)}
+     * When: Try to update the {@link Variant}'s deleted attribute
+     * Should: Update it in Data base.
+     *
+     * @throws DotDataException
+     */
+    @Test
+    public void updateDeletedField() throws DotDataException {
+        final Variant variant = new VariantDataGen().next();
+
+        final Variant variantSaved = FactoryLocator.getVariantFactory().save(variant);
+
+        assertNotNull(variantSaved);
+        assertNotNull(variantSaved.getIdentifier());
+        assertFalse(variantSaved.isArchived());
+
+        final Variant variantUpdated = new Variant(variantSaved.getIdentifier(),
+                variantSaved.getName(), true);
+
+        FactoryLocator.getVariantFactory().update(variantUpdated);
+
+        final Variant variantFromDataBase = getVariantFromDataBase(variantSaved);
+
+        assertEquals(variantUpdated.getName(), variantFromDataBase.getName());
+        assertEquals(variantUpdated.getIdentifier(), variantFromDataBase.getIdentifier());
+        assertTrue(variantFromDataBase.isArchived());
+    }
+
+    /**
+     * Method to test: {@link VariantFactory#update(Variant)}
+     * When: Try to update a {@link Variant} object with a Null name
+     * Should: throw a {@link NullPointerException}
```

Comment:
We don't need this test anymore because We are using Immutable now and it does not allow to create the Variant with a null name

https://github.com/dotCMS/core/pull/22839/commits/8aaaf9e8f7d06c71edfd820737a9fcabc12d94ed#diff-5e743d42b5d325d8137e5e4a8f2efe1b0d237033cba6cfa4f66ec1b30d7e283cL239